src/main/java/com/swirlds/recordserver/TokensService.java

```diff
@@ -239,4 +241,210 @@ private void getTokenById(ServerRequest request, ServerResponse response) {
+        final JsonArrayBuilder balancesArray = JSON.createArrayBuilder();
+        for (int i = 0; i < resultTableResultSet.getRowCount(); i++) {
+            final long accountNum = resultTableResultSet.getLong(i, 0);
+            highestAccountNumber = (direction.equalsIgnoreCase("asc") ? Math.max(highestAccountNumber, accountNum) :
+                    Math.min(highestAccountNumber, accountNum));
+            balancesArray.add(JSON.createObjectBuilder()
+                    .add("account", "0.0." + accountNum)
+                    .add("balance", resultTableResultSet.getLong(i, 2))
+                    .add("tokens",JSON.createArrayBuilder().build())
+                    .build());
+        }
+
+        final JsonObjectBuilder returnObject = JSON.createObjectBuilder()
+                .add("timestamp", (resultTableResultSet.getRowCount() <= 0 ? 0
+                        : (long)resultTableResultSet.getDouble(0, 3)))
+                .add("balances", balancesArray.build());
+        if (!singleAccountMode) {
+            String nextLink = "/api/v1/tokens/" + tokenId + "/balances?order=" + direction + "&limit=" + limit +
+                    "&account.id=" + (direction.equalsIgnoreCase("asc") ? "gt" : "lt") + ":0.0." + highestAccountNumber;
+            returnObject.add("links", JSON.createObjectBuilder()
+                        .add("next", nextLink)
+                        .build());
+        }
+        response.send(returnObject.build());
+    }
+
+/**********************
+### listNfts specifying only a token id
+GET http://localhost:8080/api/v1/tokens/629591/nfts?limit=99
+### listNfts specifying a token id and an account id
+GET http://localhost:8080/api/v1/tokens/629591/nfts?account.id=644972
+### listNfts specifying a token id and a range of account ids
+GET http://localhost:8080/api/v1/tokens/629591/nfts?account.id=gt:644000&account.id=lt:644999&limit=1000
+### listNfts specifying a token id and a range of account ids, sorted in descending order
+GET http://localhost:8080/api/v1/tokens/629591/nfts?account.id=gt:644000&account.id=lt:644999&limit=1000&order=desc
+ **********************/
+    private void listNfts(ServerRequest request, ServerResponse response) {
+        final Optional<String> accountIdQueryParam = request.queryParams().first("account.id");
+        final Optional<String> limitParam = request.queryParams().first("limit");
+        final Optional<String> orderParam = request.queryParams().first("order");
+
+        // restrict query to only the tokenId passed in as a path parameter
+        final String tokenId = request.path().segments().get(0);
+
+        // build and execute query
+        final List<QueryParamUtil.WhereClause> whereClauses = new ArrayList<>();
+        whereClauses.add(QueryParamUtil.parseQueryString(QueryParamUtil.Type._long, "token_id",
+                request.path().segments().get(0)));
+        accountIdQueryParam.ifPresent(s ->
```

Comment:
similarly here. account_id has other operators, not just equals.

Reply:
Same as above as well -- it handles =, <. <=, >, and >= for sure (and potentially != as well).